hail/src/main/scala/is/hail/expr/ir/EmitClassBuilder.scala

```diff
@@ -953,44 +940,16 @@ class EmitMethodBuilder[C](
           }
         }
 
-        new EmitValue {
-          evSelf =>
-
-          override def emitType: EmitType = emitCode.emitType
-
-          override def load: EmitCode = emitCode
-
-          override def get(cb: EmitCodeBuilder): SCode = emitCode.toI(cb).get(cb)
-        }
+        EmitValue(if (required) None else Some(mb.getArg[Boolean](codeIndex + 1)), sct.loadToSValue(cb, null, mb.getArg(codeIndex)(sct.ti)))
 
       case SCodeEmitParamType(et) =>
         val ts = et.st.codeTupleTypes()
 
-        new EmitValue {
-          evSelf =>
-          val emitType: EmitType = et
-
-          def load: EmitCode = {
-            EmitCode(Code._empty,
-              if (et.required)
-                const(false)
-              else
-                mb.getArg[Boolean](codeIndex + ts.length),
-              st.fromCodes(ts.zipWithIndex.map { case (t, i) =>
-                mb.getArg(codeIndex + i)(t).get
-              }))
-          }
-
-          override def get(cb: EmitCodeBuilder): SCode = {
-            new SValue {
-              override def get: SCode = st.fromCodes(ts.zipWithIndex.map { case (t, i) =>
-                mb.getArg(codeIndex + i)(t).get
-              })
-
-              override def st: SType = evSelf.st
-            }
-          }
-        }
+        val m = if (et.required) None else Some(mb.getArg[Boolean](codeIndex + ts.length))
+        val v = et.st.fromValues(ts.zipWithIndex.map { case (t, i) =>
```

Comment:
I'm getting an error here with a `SIndexablePointer`. Should that have been a `SingleCodeEmitParamType`?